src/util/requestUtils.ts

```diff
@@ -57,7 +57,10 @@ export interface WebsocketClientOptions extends WSClientConfigurableOptions {
   reconnectTimeout: number;
 };
 
-export type GenericAPIResponse = Promise<any>;
+export type GenericAPIResponse<T> = Promise<{
+  success: boolean;
+  result: T;
```

Comment:
`GenericApiResponse` should be a union type that models both the `Success` and `Failure` case.

When `success` is `false`, there is no `result`.

Reply:
Agree. GenericAPIResponse was primarily intended as a placeholder that all endpoints return `Promise<semiStructuredSomething>` without being as frustrating as returning `unknown` (frustrating for those less typescript exposed). I like the suggestion of the union between both cases

Reply:
Agree. `Promise` shouldn't be part of the response type.